tequilapi/http_api_server.go

```diff
@@ -0,0 +1,74 @@
+package tequilapi
+
+import (
+	"errors"
+	"fmt"
+	"net"
+	"net/http"
+	"strconv"
+	"strings"
+	"sync"
+)
+
+type ApiServer interface {
+	Port() int
+	Wait()
+	Stop()
+}
+
+type apiServer struct {
+	listener  net.Listener
+	stopped   *sync.WaitGroup
+	boundPort int
+}
+
+func NewServer(address string, port int, handler http.Handler) (ApiServer, error) {
```

Comment:
What about splitting 2 different responsibilities:
 - `NewServer` factories  `ApiServer` struct
- `ApiServer.Start()` - starts server

Reply:
Will think about it

Reply:
Renamed to StartNewServer to reflect function intent.